src/Games/NexusMods.Games.StardewValley/WebAPI/SMAPIWebApi.cs

```diff
@@ -0,0 +1,118 @@
+using Microsoft.Extensions.Logging;
+using NexusMods.Paths;
+using StardewModdingAPI.Toolkit;
+using StardewModdingAPI.Toolkit.Framework.Clients.WebApi;
+using StardewModdingAPI.Toolkit.Utilities;
+
+namespace NexusMods.Games.StardewValley.WebAPI;
+
+/// <summary>
+/// Implementation of <see cref="ISMAPIWebApi"/> using <see cref="WebApiClient"/>
+/// </summary>
+internal sealed class SMAPIWebApi : ISMAPIWebApi
+{
+    private const string ApiBaseUrl = "https://smapi.io/api";
+    private const string NexusModsBaseUrl = "https://nexusmods.com/stardewvalley/mods";
+
+    private readonly ILogger<SMAPIWebApi> _logger;
+
+    private bool _isDisposed;
+    private WebApiClient? _client;
+
+    private readonly Dictionary<string, Uri> _knownModPageUrls = new(StringComparer.OrdinalIgnoreCase);
```

Comment:
Could this cause issues if changed while being iterated in another thread?

Reply:
Yes and no. Yes, because it's _documented_ to not be threadsafe, no because we probably won't be referencing this from multiple threads at once.

Reply:
I`d prefer to be safe than sorry. I also think that the chances of a data race aren't high, but then again I'm sure there are lots of developers that thought so as well just to run into hard to debug issues later on.

Reply:
Resolved using an atomic `ImmutableDictionary`. This makes the most sense for the very read heavy behavior.